artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -291,6 +291,7 @@ public void applySetting(final AddressSettings addressSettings) {
 
       if (pageLimitBytes != null && pageSize > 0) {
          estimatedMaxPages = pageLimitBytes / pageSize;
+         checkNumberOfPages();
```

Comment:
Calling this here would have it act before start() in the constructor case, which seems like it could lead to duplicate logging.

In the non-constructor case it seems like it could cause logging every time any [possibly-unrelated] setting changes, which may also not be desirable. Maybe it could check if any of those settings changed?